MAINTAINING.md

```diff
@@ -49,6 +49,70 @@ Note `--distribution-type all`.
 
 ### New IDE version
 
+Each new major platform release brings not only a lot of new features
+but usually also a bunch of new incompatibilities (both source and binary ones).
+Since we want to support at least the latest stable major platform release and new [EAP](https://www.jetbrains.com/resources/eap/)s,
+we have to compile the plugin with different platform versions to avoid any runtime errors.
+
+To solve the problem with incompatibilities, we use conditional compilation based on Gradle [source sets](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSet.html).
+See [CONTRIBUTING.md](CONTRIBUTING.md#source-code-for-different-platform-versions) for more details about concrete source structure.
+
+#### How to support new platform version
+
+The following explanation uses `old`, `current` and `new` platform terms that mean:
+* `old` - number of the oldest supported platform version that should be dropped while
+* `current` - number of the latest major stable platform release
+* `new` - number of new platform that should be supported
+
+For example, at the moment of writing, `201` is `old`, `202` is `current` and `203` is `new`.
+ 
+Step to step instruction how to support new platform version:
```

Comment:
```suggestion
Step by step instruction on how to support new platform version:
```

Reply:
Not sure about preposition here

Reply:
<img src="https://user-images.githubusercontent.com/6079006/94683124-abc31580-032e-11eb-9225-1212fbe36035.png" height="200">

<img src="https://user-images.githubusercontent.com/6079006/94683150-b4b3e700-032e-11eb-9358-0f89173dd2e0.png" height="200">